### PR TITLE
Refactor the Vanilla ES6 example

### PR DESCRIPTION
**Pull request by xorgy**
I refactored the Vanilla ES6 to be cleaner and more idiomatic. I think it now represents it more fairly.

I also added a secton to learn.json for it, set up the learn sidebar, and added a link to it on the front page.

### Changelog

 - Simplify the Template module.
 - Simplify the router.
 - Convert all of the View render commands to normal methods, and give them more meaningful names.
 - Convert all of the View bind synthetic events to binding methods, and give them more meaningful names.
 - Normalize the Store API.
 - Collapse the Model abstraction, since it consisted mainly of trampolines to methods on Store.
 - Remove unnecessary dynamic templating for the Clear completed button.
 - Put .footer inside .main since they are always hidden together.
 - Make .filter a div, and unnest the <a>s inside the <li>s to avoid unnecessary extra nodes and CSS.
 - Update the tests to work correctly with the new Vanilla ES6 markup.
 - Remove unnecessary object orientation in app.js.
 - Remove boolean traps from methods.
 - Do not modify the NodeList prototype.
 - Use cross-browser copmatible NodeList iteration in dispatchEvent of $delegate.
 - Fix the existing JSDoc comments.
 - Expand the existing JSDoc comments.
 - JSDoc more of the code.
 - Use Google Closure Compiler instead of Babel. It offers useful warnings and generates better code.
 - Author a learn.json section for the Vanilla ES6 example.

### CODE REVIEW

**Comment by addyosmani** [comment]
@xorgy Would you mind rebasing this against master? Sorry for the delay in getting this reviewed.

**Comment by xorgy** [comment]
Sure thing :- )

**Comment by xorgy** [comment]
Seems like the test runner is not connecting to webdriver on Travis, not sure what's up with that. Definitely works locally.

**Comment by samccone** [comment]
> Seems like the test runner is not connecting to webdriver on Travis, not sure what's up with that. Definitely works locally.


Yeah just need to find some time to debug what is happening :goat:

**Comment by samccone** [comment]
boom, looks good! thanks for the PR

File: examples/vanilla-es6/src/store.js [hunk 1]
```diff
@@ -52,93 +65,90 @@ export default class Store {
...
-	 * Will drop all storage and start fresh
+	 * Count total, active, and completed todos.
 	 *
-	 * @param {function} callback The callback to fire after dropping the data
+	 * @param {function(number, number, number)} callback Called when the count is completed
 	 */
-	drop(callback){
-		localStorage[this._dbName] = JSON.stringify({todos: []});
+	count(callback) {
+		this.find(emptyItemQuery, data => {
```
Comment (avinashdvv) [inline examples/vanilla-es6/src/store.js:142]: @xorgy here why not directly get the todos by using `this.getLocalStorage()` here

Comment (xorgy) [reply examples/vanilla-es6/src/store.js:142]: Because once you compile it, the resulting code is smaller, I believe.

**Comment by jimmywarting** [comment]
You should know by now that strict ESM require explicit path. add that missing `.js` in the end

Ryan dahl regretted ever implementing extension-less and index files. _"It's how the web works_

**Pull request merged**